src/DynamoRevit/DynamoRevit.cs

```diff
@@ -149,6 +154,22 @@ public Result Execute(ExternalCommandData commandData, ref string message, Eleme
             return Result.Succeeded;
         }
 
+        private static void Application_Idling(object sender, IdlingEventArgs e)
+        {
+            Action action = null;
+            lock (actions)
+            {
+                if (!actions.Any())
+                    return;
+
+                action = actions[0];
+                actions.RemoveAt(0);
+            }
+
+            if (action != null)
+                action();
+        }
+
```

Comment:
And this method be renamed as `OnApplicationIdle`, with the following updated implementation:

```csharp
private static void OnApplicationIdle(object sender, IdlingEventArgs e)
{
    Action pendingAction = null;
    
    lock(idleActions)
    {
        if (idleActions.Count <= 0)
            return;

        pendingAction = idleActions[0];
        idleActions.RemoveAt(0);
    }

    pendingAction(); // Execute the action.
}
```